### PR TITLE
snappy: disable auto-import of assertions on classic

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: cmd/snap/cmd_auto_import_test.go [hunk 1]
```diff
@@ -93,3 +97,26 @@ func (s *SnapSuite) TestAutoImportAssertsHappy(c *C) {
...
 	c.Check(n, Equals, total)
 }
+
+func (s *SnapSuite) TestAutoImportAssertsHappyNotOnClassic(c *C) {
+	restore := release.MockOnClassic(true)
+	defer restore()
+
+	fakeAssertData := []byte("my-assertion")
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		panic("not auto-import is run on classic")
```
Comment (mwhudson) [inline cmd/snap/cmd_auto_import_test.go]: Not sure this string makes sense.

Comment (niemeyer) [inline cmd/snap/cmd_auto_import_test.go]: Yeah, it needs tweaking. I'll merge as that's just in the test and mvo is going to be out for a few days.

**Comment by niemeyer** [comment]
Actually, is there anything running auto-import automatically?  We don't have console-conf on classic, and we're not installing the udev rule on classic, are we?  If nothing runs auto-import, it doesn't feel like we need to disable the command itself.

File: cmd/snap/cmd_auto_import_test.go (hunk 1 above)
Comment (mvo5) [inline cmd/snap/cmd_auto_import_test.go]: I improved the comment now.

**Comment by mvo5** [comment]
We currently do have the same package (including the udev rule) on classic and on core. The behavior between the two is different based on code (`if release.OnClassic { ... }`) currently, not because we install different files or rules.

This can be changed of course if you prefer that we use this approach instead of the checking in code. This means we will need to write the udev rules either in livecd-rootfs (outside of the snapd package). Or we need to create a new package "snapd-install-this-package-on-core-only" (or something similar that is shorter but conveys a similar message) and only install this package in the core snap via livecd-rootfs and not into classic.

**Pull request merged**